packages/client/src/StreamrClient.ts

```diff
@@ -106,15 +125,18 @@ class StreamrClientBase implements Context {
         options: StreamDefinition,
         onMessage?: SubscriptionOnMessage<T>
     ): Promise<Subscription<T>>
-    subscribe<T>(
+    async subscribe<T>(
         options: StreamDefinition & { resend?: ResendOptions },
         onMessage?: SubscriptionOnMessage<T>
     ): Promise<Subscription<T> | ResendSubscription<T>> {
+        let result
         if (options.resend !== undefined) {
-            return this.resendSubscribe(options, options.resend, onMessage)
+            result = await this.resendSubscribe(options, options.resend, onMessage)
+        } else {
+            result = await this.subscriber.subscribe(options, onMessage)
         }
-
-        return this.subscriber.subscribe(options, onMessage)
+        this.eventEmitter.emit('subscribe', undefined)
```

Comment:
same question as above
```suggestion
        this.eventEmitter.emit('subscribe')
```